dsc/src/main.rs

```diff
@@ -0,0 +1,518 @@
+                new_region_dir, time_f,
+            );
+        }
+
+        let output_file = format!("downstairs-{}.txt", port);
+        let output_path = {
+            let mut t = self.output_dir.clone();
+            t.push(output_file);
+            t
+        };
+
+        let dsi = DownstairsInfo {
+            ds_bin: self.rs.ds_bin.clone(),
+            region_dir: new_region_dir,
+            port,
+            _create_output: String::from_utf8(output.stdout).unwrap(),
+            output_file: output_path,
+        };
+        self.rs.ds.push(dsi);
+        Ok(time_f)
+    }
+
+    /**
+     * Delete a region directory at the given port.
+     */
+    fn delete_ds_region(&mut self, port: u32) -> Result<()> {
+        // Create the path for this region by combining the region
+        // directory and the port this downstairs will use.
+        let mut my_region = PathBuf::new();
+        my_region.push(self.rs.region_dir.clone());
+        my_region.push(format!("{}", port));
+
+        let new_region_dir =
+            my_region.clone().into_os_string().into_string().unwrap();
+
+        std::fs::remove_dir_all(&new_region_dir)?;
+
+        // If this region was part of the ds vec, remove it.
+        self.rs.ds.retain(|ds| ds.port != port);
+        Ok(())
+    }
+
+    // Start all downstairs.
+    fn start_all_downstairs(&mut self) -> Result<()> {
+        for ds in self.rs.ds.iter() {
+            let cmd = ds.clone().start().unwrap();
+            println!("started ds: {:?}", cmd);
+            self.cmd.push(cmd);
+        }
+        Ok(())
+    }
+}
+
+// WIP.. begin ignore
+fn startall(ti: &mut TestInfo) -> Result<()> {
+    let _ = ti.create_ds_region(3810, 10, 20, 4096, false).unwrap();
+    let _ = ti.create_ds_region(3820, 10, 20, 4096, false).unwrap();
+    let _ = ti.create_ds_region(3830, 10, 20, 4096, false).unwrap();
+
+    println!("All regions created, now start all downstairs");
+    ti.start_all_downstairs().unwrap();
+
+    println!("ti: {:?}", ti);
+
+    // Spawn a thread to watch each downstairs??
+    // No, ZZZ spawn a thread to first create, then wait on a downstairs.
+    // The PID should be updated somewhere another task can find it.
+    // Setup a mpsc channel and a head controller thread that knows if a
+    // thread dies unexpected, and can also restart threads (downstairs)
+    // if so desired.
+    //
+    // Messages to tell each monitor thread what to do?
+    for ads in ti.cmd.iter_mut() {
+        match ads.try_wait() {
+            Ok(Some(status)) => println!("exited with: {}", status),
+            Ok(None) => {
+                println!("status not ready yet, lets really wait");
+                let res = ads.wait();
+                println!("result: {:?}", res);
+            }
+            Err(e) => println!("error attempting to wait: {}", e),
+        }
+    }
+    Ok(())
+}
+
+fn _create(cleanup: bool, extent_count: u64, extent_size: u64) -> Result<()> {
```

Comment:
still needed?

Reply:
not still.. not needed yet...
I'll remove for this PR